handlers.go

```diff
@@ -172,9 +184,17 @@ func handleUnits(add func(Change), services map[string]*charm.ServiceSpec, added
 			if i < numPlaced {
 				p = service.To[i]
 			}
+			var series = service.Series
+			if series == "" {
+				serviceCharm := charm.MustParseURL(service.Charm)
```

Comment:
Same comment here. Since this logic is repeated twice, is it worth an external function? Like
```go
func getSeries(service *charm.ServiceSpec, defaultSeries string) string {
	if service.Series != "" {
		return service.Series
	}
	// The following is safe because the bundle data is assumed to be already
	// verified, and therefore this must be a valid charm URL.
	series := charm.MustParseURL(service.Charm).Series
	if series != "" {
		return series
	}
	return defaultSeries
}
```